### PR TITLE
feather: init at 2.5.2

### PR DESCRIPTION
**Pull request by surfaceflinger**
## Description of changes

Added a package for [Feather Wallet](https://featherwallet.org/). Based on #186382 which seems stale now.

 - [x] Works with my Ledger
 - [x] Tested with my own Monero node
 - [x] Tested embedded TOR and connected to a remote onion node
 - [x] Made a churn transaction

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
I don't have a Monero wallet, so I cannot confirm full functionality, but the GUI launches on x86_64-linux X11 & aarch64-linux Wayland. Some suggestions.

File: pkgs/by-name/fe/feather/package.nix [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    "-DReadline_LIBRARY=${readline}/lib/libreadline.so"
+    "-DReadline_ROOT_DIR=${readline.dev}"
+    "-DTOR_DIR=${tor}/bin"
+    "-DTOR_VERSION=${tor.version}"
+  ];
+
+  qtWrapperArgs = with qt6; [
+    "--set QT_PLUGIN_PATH ${qtwayland}/${qtbase.qtPluginPrefix}"
+    "--set QT_QPA_PLATFORM_PLUGIN_PATH ${qtwayland}/${qtbase.qtPluginPrefix}/platforms"
+  ];
```
[the remark this answers is not in the document]
Comment (OPNA2608) [reply pkgs/by-name/fe/feather/package.nix:67]: Build is now done. Adding `qtwayland` to `buildInputs` and removing these seems to work well enough under Miriway. Can you confirm this under whatever setup you're using?

[the remark this answers is not in the document]
Comment (surfaceflinger) [reply pkgs/by-name/fe/feather/package.nix:67]: oh, weird, didn't work for me as far as I remember, I'll check again after work today

**Review by OPNA2608 — APPROVED** [review]
Result of `nixpkgs-review pr 265694` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>feather</li>
  </ul>
</details>

---

Some things I wouldn't say are worth blocking over, but perhaps still good to mention / talk about.

CMake says there are disabled tests which could be enabled, but [upstream says](https://github.com/feather-wallet/feather/blob/135f1987c9548372a111c276962ac1a67ddc00c7/MAINTENANCE.md)

> Feather does not currently have a test suite (apart from the tests in the Monero submodule), this is a WIP.

…so not sure if enabling them makes sense, or how much effort it would take to get them working.

File: pkgs/by-name/fe/feather/package.nix [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  cmakeFlags = [
+    "-DProtobuf_INCLUDE_DIR=${lib.getDev protobuf}/include"
+    "-DProtobuf_PROTOC_EXECUTABLE=${lib.getExe protobuf}"
+    "-DReadline_INCLUDE_DIR=${lib.getDev readline}/include/readline"
+    "-DReadline_LIBRARY=${lib.getLib readline}/lib/libreadline.so"
+    "-DReadline_ROOT_DIR=${lib.getDev readline}"
+    "-DTOR_DIR=${lib.makeBinPath [ tor ]}"
+    "-DTOR_VERSION=${tor.version}"
+  ];
+
```
[the remark this answers is not in the document]
Comment (surfaceflinger) [reply pkgs/by-name/fe/feather/package.nix:67]: That's so cool, thanks

**Review by OPNA2608 — APPROVED** [review]
Stuck on the weak ARM hardware so can't test building, but OfBorg is fine & the version test runs so LGTM!

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 265694` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>feather</li>
  </ul>
</details>

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

LGTM, thanks!

**Pull request merged**